plugins/woocommerce-admin/client/products/fields/options/options.tsx

```diff
@@ -24,17 +25,24 @@ export const Options: React.FC< OptionsProps > = ( {
 	const { values } = useFormContext< Product >();
 	const { generateProductVariations } = useProductVariationsHelper();
 
-	const handleChange = async ( attributes: ProductAttribute[] ) => {
-		onChange( attributes );
-		generateProductVariations( { ...values, attributes } );
-	};
+	const { attributes, handleChange } = useProductAttributes( {
+		allAttributes: value,
+		isVariationAttributes: true,
+		onChange: ( newAttributes ) => {
```

Comment:
Depending on if your wrapping the `handleChange` in a `useCallback` we should also wrap this `onChange` in a `useCallback` as it will be one of the dependencies of `handleChange`.

Reply:
I started to wrap this in a `useCallback` but shortly after realized that we need the most up-to-date `values` inside this function, so this would get recomputed every time.  Meaning that the `useCallback` would result in worse performance, so I opted not to add this in.

Reply:
Aaah yeah that makes sense, I initially missed that. Scratch this then!